src/plugins/conditionals/conditionals.c

```diff
@@ -475,34 +497,65 @@ static const char * isAssign (char * expr, Key * parentKey, KeySet * ks)
 		--lastPtr;
 	if (*firstPtr != '\'' || *lastPtr != '\'')
 	{
-		if (lastPtr <= firstPtr) return NULL;
+		if (lastPtr <= firstPtr)
+		{
+			ELEKTRA_SET_ERRORF (134, parentKey,
+					    "Invalid syntax: \"%s\". Check kdb info conditionals for additional information\n", expr);
+			return NULL;
+		}
 		*(lastPtr + 1) = '\0';
-		Key * lookupKey = keyDup (parentKey);
-		keyAddName (lookupKey, firstPtr);
+		Key * lookupKey;
+		if (*firstPtr == '@')
+		{
+			lookupKey = keyDup (parentKey);
+			++firstPtr;
+			keyAddName (lookupKey, firstPtr);
+		}
+		else if (!strncmp (firstPtr, "..", 2) || !strncmp (firstPtr, ".", 1))
+		{
+			lookupKey = keyDup (key);
+			keyAddName (lookupKey, firstPtr);
+		}
+		else
+		{
+			lookupKey = keyNew (firstPtr, KEY_END);
+		}
 		Key * assign = ksLookup (ks, lookupKey, KDB_O_NONE);
-		keyDel (lookupKey);
 		if (!assign)
 		{
+			ELEKTRA_SET_ERRORF (133, parentKey, "Key %s not found", keyName (lookupKey));
+			keyDel (lookupKey);
 			return NULL;
 		}
 		else
 		{
+			keyDel (lookupKey);
 			return keyString (assign);
 		}
 	}
 	else
 	{
-		if (firstPtr == lastPtr) return NULL;
+		if (firstPtr == lastPtr)
+		{
+			ELEKTRA_SET_ERRORF (134, parentKey,
+					    "Invalid syntax: \"%s\". Check kdb info conditionals for additional information\n", expr);
+			return NULL;
+		}
 		char * nextMark = strchr (firstPtr + 1, '\'');
-		if (nextMark != lastPtr) return NULL;
+		if (nextMark != lastPtr)
+		{
+			ELEKTRA_SET_ERRORF (134, parentKey,
+					    "Invalid syntax: \"%s\". Check kdb info conditionals for additional information\n", expr);
```

Comment:
Should these errors really have the same reason? Can you give the user some more specific information?